app/src/main/java/com/moscow/tudee/presentation/screen/category/categoriesScreen/CategoryViewModel.kt

```diff
@@ -0,0 +1,121 @@
+package com.moscow.tudee.presentation.screen.category.categoriesScreen
+
+import com.moscow.tudee.R
+import com.moscow.tudee.domain.entity.Category
+import com.moscow.tudee.domain.service.CategoryServices
+import com.moscow.tudee.presentation.BaseViewModel
+import com.moscow.tudee.presentation.screen.category.categoriesScreen.CategoriesEvents
+import com.moscow.tudee.presentation.screen.category.CategoriesScreenState
+import com.moscow.tudee.presentation.screen.category.toCategory
+import com.moscow.tudee.presentation.screen.category.toCategoryUi
+
+class CategoryViewModel(
+    private val categoryServices: CategoryServices,
+) : BaseViewModel<CategoriesScreenState, CategoriesEvents>(CategoriesScreenState()),
+    CategoriesInteractionListener {
+
+    init {
+        getAllCategories()
+    }
+
+    fun onReturnedFromEditWithMessage(messageResId: Int?) {
+        getAllCategories()
+        messageResId?.let {
+            updateState {
+                it.copy(
+                    successMessage = messageResId,
+                    isSnackBarShow = true
+                )
+            }
+        }
+    }
+
+    private fun getAllCategories() {
+        launchWithResult(
+            action = { categoryServices.getCategories() },
+            onSuccess = ::onGetAllCategoriesSuccess,
+            onError = ::onGetAllCategoriesFailed,
+            onStart = ::onLoading,
+            onFinally = ::onFinally
+        )
+    }
+
+    private fun onGetAllCategoriesSuccess(categories: List<Category>) {
+        val categoryUiList = categories.map { it.toCategoryUi() }
+        updateState {
+            it.copy(
+                categories = categoryUiList,
+            )
+        }
```

Comment:
make it in one line, but it still readable if not changed.
```suggestion
        updateState { it.copy( categories = categoryUiList) }
```